docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,669 @@
+
+ASO v1 generated secrets on the users behalf in some cases. The users had access to the generated secret as ASO wrote it into a Kubernetes (or KeyVault) 
+secret.
+
+There are problems with this approach:
+1. It makes performing secret rollover more difficult. If the user had specified the secret, rollover is as easy as modifying the specified secret, which 
+   triggers a reconcile, which triggers a PUT to the Azure Resource updating the secret. If the operator generated the secret then the user must somehow 
+   issue an instruction to us to perform an action to roll the secret, which doesn't fit as well into the goal-seeking paradigm Kubernetes prefers.
+2. It takes control away from the user. What if the secret the operator generates doesn't comply with a particular organizations complexity requirements?
+3. It requires that we do a _very_ good job generating cryptographically secure passwords/secrets. This could easily become a can of worms.
+4. It hinders adoption of existing resources. If the operator expects to always generate the secret for a SQL DB, but the user wants to import a SQL DB 
+   they've already created through some other mechanism, then they are at an impasse as there is no (easy) way for them to provide ASO with the secret.
+5. It doesn't work well with GitOps. A number of customers have expressed the desire to move resources between namespaces. In theory this is easy - 
+   just create the exact same resources in a different namespace (pointing at the same Azure resources), mark the old resources as `skip-deletion`, and 
+   delete the old resources. If the definition of the secret isn't part of the users GitOps flow this becomes more difficult as the first namespace has 
+   the secret (which was created by ASO) and that secret must be cloned manually to the second namespace, otherwise the creation of the resource in the 
+   second namespace will attempt to generate a new secret.
+
+We could investigate allowing both automatically generated secrets (the default) and user specified secrets (opt-in). That avenue results in the operator 
+shouldering all of the complexity burden of both though. Given ASO's approach as a low level toolkit providing the ability to create Azure resources, at least 
+at this time we should avoid the added complexity of generating user secrets.
+
+### Prefer Managed Identity
+
+Managing secrets is hard. We should prefer AAD Authentication and Managed Identity where possible.
+
+What this means in practice is that secrets should not be retrieved **unless the user has specifically asked for them to be**.
+
+This has a number of advantages:
+1. If the user _is_ using Managed Identity, they don't have a secret leak waiting to happen created in their namespace which they didn't ask about 
+   (and might not even know about).
+2. If the user _is_ using Managed Identity, many services support ways to instruct the `ListKeys` (or equivalent) API to error when called, as a way to
```

Comment:
> Are you just saying we should only call ListKeys at the user's behest (controlled by some for-operator part of the spec), so that we can interpret that error as fatal?

Yes. I've restructured this sentence quite a bit to make the important part come first.